Sources/Moya/MoyaProvider+Internal.swift

```diff
@@ -83,18 +84,57 @@ public extension MoyaProvider {
             switch stubBehavior {
             case .never:
                 switch target.task {
-                case .request:
+                case .requestData(let data):
+                    preparedRequest.httpBody = data
+                    cancellableToken.innerCancellable = self.sendRequest(target, request: preparedRequest, queue: queue, progress: progress, completion: networkCompletion)
+
+                case let .requestEncoded(parameters: parameters, encoding: parameterEncoding):
+                    do {
```

Comment:
Sorry, I don't really understand how I could throw a `MoyaError.requestMapping` from within the `performNetworking` closure, also I have no idea how to move the code to the endpoint. How can I switch on the `target.task` from within the Endpoint class? Need some help with this ...